lib/sidekiq_rx_impl.cc

```diff
@@ -136,31 +150,41 @@ bool sidekiq_rx_impl::stop() {
 uint8_t sidekiq_rx_impl::get_rx_gain_mode() {
 	skiq_rx_gain_t result;
 
+    /* gain mode is same for both channels */
 	if (skiq_read_rx_gain_mode(card, hdl, &result) != 0) {
 		printf("Error: could not get gain mode\n");
+		this->stop();
 	}
 	return static_cast<uint8_t >(result);
 }
 
 void sidekiq_rx_impl::set_rx_gain_mode(uint8_t value) {
+    if(dual_channel) {
+
+        if (skiq_write_rx_gain_mode(card, hdl2, static_cast<skiq_rx_gain_t>(value)) != 0) {
+            printf("Error: could not set gain mode to %d\n", value);
+        }
+    }
 	if (skiq_write_rx_gain_mode(card, hdl, static_cast<skiq_rx_gain_t>(value)) != 0) {
 		printf("Error: could not set gain mode to %d\n", value);
 	}
 }
 
-double sidekiq_rx_impl::get_rx_gain() {
+double sidekiq_rx_impl::get_rx_gain(int handle) {
         double cal_offset;
 
-	if (skiq_read_rx_cal_offset(card, hdl, &cal_offset) != 0) {
+	if (skiq_read_rx_cal_offset(card, (skiq_rx_hdl_t)handle, &cal_offset) != 0) {
 		printf("Error: could not get calibration offset\n");
+		this->stop();
```

Comment:
should this be considered fatal and result in stop() being called?

Reply:
I think we need to agree on a philosophy.  In all cases if I get a skiq error I should leave gracefully.  Except when there are cases that an error can occur due to a card not having a feature or something.

So is there a case where I can get an error here and it is expected?

Reply:
In general, I believe GNU radio recommends using exceptions for errors and I believe that we should try to stick to that.  This provides users of the gr-sidekiq block to determine how they'd like to handle the error, they may want to perform additional handling of the state prior to the exit being called.  For some info on the recommended error handling, you can refer to https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#S-errors.  

As far as whether there is a known case when this error could occur, there is nothing that immediately comes to mind.  However, I agree that a general philosophy needs to be adopted and adhered to.

Reply:
Ok, I am really confused.  Should I call this->stop() when a fatal error occurs?  Or just return? I don't see any of the other code throwing some exception, should I implement throwing an exception?